skimage/transform/_warps.py

```diff
@@ -127,26 +132,21 @@ def resize(image, output_shape, order=None, mode='reflect', cval=0, clip=True,
             elif np.any((anti_aliasing_sigma > 0) & (factors <= 1)):
                 warn("Anti-aliasing standard deviation greater than zero but "
                      "not down-sampling along all axes")
-
-        # Translate modes used by np.pad to those used by ndi.gaussian_filter
-        np_pad_to_ndimage = {
-            'constant': 'constant',
-            'edge': 'nearest',
-            'symmetric': 'reflect',
-            'reflect': 'mirror',
-            'wrap': 'wrap'
-        }
-        try:
-            ndi_mode = np_pad_to_ndimage[mode]
-        except KeyError:
-            raise ValueError("Unknown mode, or cannot translate mode. The "
-                             "mode should be one of 'constant', 'edge', "
-                             "'symmetric', 'reflect', or 'wrap'. See the "
-                             "documentation of numpy.pad for more info.")
-
         image = ndi.gaussian_filter(image, anti_aliasing_sigma,
                                     cval=cval, mode=ndi_mode)
 
+    if NumpyVersion(scipy.__version__) >= '1.6.0':
+        # The grid_mode kwarg was introduced in SciPy 1.6.0
+        order = _validate_interpolation_order(image.dtype, order)
+        zoom_factors = [1 / f for f in factors]
+        image = convert_to_float(image, preserve_range)
+        out = ndi.zoom(image, zoom_factors, order=order, mode=ndi_mode,
+                       cval=cval, grid_mode=True)
+        _clip_warp_output(image, out, order, mode, cval, clip)
+        return out
+
+    # TODO: Remove the fallback code below once SciPy >= 1.6.0 is required.
+
     # 2-dimensional interpolation
     if len(output_shape) == 2 or (len(output_shape) == 3 and
```

Comment:
Simply to avoid an extra `return` statement.
```suggestion

    # TODO: Remove the fallback code below once SciPy >= 1.6.0 is required.

    # 2-dimensional interpolation
    elif len(output_shape) == 2 or (len(output_shape) == 3 and
```